src/lib.rs

```diff
@@ -390,10 +396,10 @@ impl MeasurementDuration for sensor_class::Sht1Gen {
     /// Maximum measurement duration:
     /// - Normal mode: 14.4 ms (SHTC1/SHTW2 datasheet 3.1)
     /// - Low power mode: 0.94 us (SHTC1/SHTW2 low power application note)
-    fn max_measurement_duration(mode: PowerMode) -> u16 {
+    fn max_measurement_duration(mode: PowerMode) -> u32 {
         match mode {
-            PowerMode::NormalMode => 14400,
-            PowerMode::LowPower => 940,
```

Comment:
After this change, the docs and the implementation don't match anymore. But in any case, there's no need for a change, `DelayNs` also provides a `delay_us` method (and we don't need nanosecond precision).